### ISSUE TITLE
[Bug] BepInEx Gets reinstalled every Deploy although already installed

### ISSUE THREAD

**Issue by Der-Floh** [Nexus-Mods/vortex-api#38]
In my own Extension Script:
I'm having the problem that whenever Vortex tries to Deploy Mods, it also tries to install BepInEx although it is already installed and Enabled. Continuing with replacing the installed BepInEx actually does replace the Mod but after that triggers the next Deployment in which the same thing happens again.

![Vortex-BepInEx-DeployBug](https://github.com/Nexus-Mods/vortex-api/assets/65826571/233ef332-3fd2-4295-9b32-a0fe5a8fbc53)

https://github.com/Nexus-Mods/vortex-api/assets/65826571/7c4d92a3-f894-4a32-b3db-5c1cd87f94c0

My main function currently looks like this
```javascript
function main(context) {
  context.requireExtension("modtype-bepinex");
  context.registerGame({
    id: GAME_ID,
    name: GAME_NAME,
    mergeMods: true,
    queryPath: findGame,
    queryModPath: () => "",
    logo: "gameart.jpg",
    executable: () => GAME_EXE,
    requiredFiles: [GAME_EXE],
    setup: prepareForModding,
    environment: { SteamAPPId: STEAM_APP_ID },
    details: { steamAppId: STEAM_APP_ID },
  });

  context.once(() => {
    if (context.api.ext.bepinexAddGame !== undefined) {
      context.api.ext.bepinexAddGame({ gameId: GAME_ID, autoDownloadBepInEx: true });
    }
  });

  context.registerInstaller("mod", 25, testSupportedContent, installContent);

  return true;
}
```

Why does this happen? I couldn't find any other one that had this.

I'm on Vortex 1.9.5 Windows 11

**Comment by IDCs** [comment]
How odd - can you please double click the BepInEx mod and ensure that it's assigned the Bepis Injector modType?

If it is, can you please send us your full script so we can run some tests?

FYI - we have an internal ticket for our bepinex extension - basically we're going to run a few tests and improve it; your script might help us pinpoint certain issues we were not aware of.

**Comment by Der-Floh** [comment]
The Mod Type is actually BepInEx (Plugins)
![Vortex-BepInEx-DeployBug-ModType](https://github.com/Nexus-Mods/vortex-api/assets/65826571/e2bd25e7-ccf7-43fb-acd5-9b1bdfa18bda)
But it should be Bepis Injector Extensible?

**Comment by Der-Floh** [comment]
If I set it to Bepis Injector I no longer get the loop but I get a Message stating that the Mod is Redundant
![Vortex-BepInEx-DeployBug-Redundant](https://github.com/Nexus-Mods/vortex-api/assets/65826571/37e52eba-5628-4844-a5cc-019a5a535d4c)

If I click on Review (after BepInEx Mod Filename):
![Vortex-BepInEx-DeployBug-Redundant-Review](https://github.com/Nexus-Mods/vortex-api/assets/65826571/d41c6177-0d9c-4619-af6a-e0fa9edbc351)

**Comment by IDCs** [comment]
It should indeed be assigned to the Bepis Injector modType. It looks like Vortex thinks there's already a mod that deploys the same files to the game's folder. Alternatively the installed mod folder is empty (doubt it)

Try deleting the mod (and archive) and hit deploy again.

**Comment by Der-Floh** [comment]
Uppon deleting Mod and Archive after it got redownloaded (as it should). I had to change the Mod Type again (it was BepInEx Plugins again, apparantly I have to change that every time, I guess that also shouldnt be the case) but then it worked. But, there is still a problem, when looking into the Mod Staging Folder there is now the following structure:
```
MyCustomGame
|--> BepInEx_x64_5.4.17.0
     |--> BepInEx
          |--> plugins
               |-->BepInEx.dll (and the other dll files)
```
Shouldn't the Folder in which the BepInEx dll files are be the `core` Folder and not the `plugins` folder? (When looking into BepInEx on NexusMods there it also is the core folder) I guess because the Mod Type was Plugins on download / installation?

**Comment by Der-Floh** [comment]
@IDCs Do you know how to fix this? Or what could I try to do?

**Comment by IDCs** [comment]
Hi @Der-Floh, apologies for the late response, I wasn't notified of your comment for some reason. Can you send me a friend request on Discord so that we can debug this together. My username is nagev_

**Comment by Der-Floh** [comment]
Quick summary of what was the Problem:
Appearently the the BepInEx installer Priority is 50 (at this time) but my custom installer had 25 (lower comes first). So my installer was triggert before the BepInEx one which caused these issues.

**Comment by IDCs** [comment]
I've reduced the installer priority to 10 as part of my extension review - this update should be available in future releases of Vortex (1.9.11 =<)

https://github.com/Nexus-Mods/extension-modtype-bepinex/commits/184_extension_review

Thanks again for helping us to discover this issue.
